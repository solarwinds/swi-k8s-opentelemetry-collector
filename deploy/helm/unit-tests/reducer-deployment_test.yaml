# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for reducer-deployment
templates:
  - network/reducer-deployment.yaml
  - network/configmap.yaml
tests:
  - it: Deployment spec should match snapshot when ebpfNetworkMonitoring is enabled
    template: network/reducer-deployment.yaml
    set:
      ebpfNetworkMonitoring.enabled: true
    asserts:
      - matchSnapshot:
          path: spec.template.spec
  - it: Deployment should not be generated when ebpfNetworkMonitoring is disabled
    template: network/reducer-deployment.yaml
    set:
      ebpfNetworkMonitoring.enabled: false
    asserts:
      - hasDocuments:
          count: 0
