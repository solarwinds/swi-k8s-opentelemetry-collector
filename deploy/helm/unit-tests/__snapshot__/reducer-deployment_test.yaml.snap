Deployment spec should match snapshot when ebpfNetworkMonitoring is enabled:
  1: |
    containers:
      - args:
          - --port=7000
          - --log-console
          - --no-log-file
          - --warning
          - --enable-aws-enrichment
          - --disable-prometheus-metrics
          - --enable-otlp-grpc-metrics
          - --otlp-grpc-metrics-host=RELEASE-NAME-swo-k8s-collector-metrics-collector
          - --otlp-grpc-metrics-port=4317
          - --num-ingest-shards=3
          - --num-matching-shards=3
          - --num-aggregation-shards=3
        env:
          - name: EBPF_NET_CRASH_METRIC_HOST
            value: RELEASE-NAME-swo-k8s-collector-metrics-collector
          - name: EBPF_NET_CRASH_METRIC_PORT
            value: "4317"
        image: solarwinds/opentelemetry-ebpf-reducer:v0.10.3
        imagePullPolicy: IfNotPresent
        name: reducer
        ports:
          - containerPort: 7000
            name: telemetry
            protocol: TCP
        readinessProbe:
          exec:
            command:
              - /srv/health_check.sh
              - readiness_probe
              - localhost
              - "7000"
          initialDelaySeconds: 5
          periodSeconds: 20
          timeoutSeconds: 5
        resources: null
        volumeMounts: null
    initContainers:
      - command:
          - sh
          - -c
          - until nc -zv $METRICS_COLLECTOR_HOST $METRICS_COLLECTOR_PORT; do echo "Waiting for metrics collector endpoint..."; sleep 5; done;
        env:
          - name: METRICS_COLLECTOR_HOST
            value: RELEASE-NAME-swo-k8s-collector-metrics-collector
          - name: METRICS_COLLECTOR_PORT
            value: "4317"
        image: busybox:1.36.1
        imagePullPolicy: IfNotPresent
        name: wait-for-metrics-collector
    nodeSelector:
      kubernetes.io/os: linux
    volumes: null
